code/game/machinery/computer/cloning.dm

```diff
@@ -15,64 +15,68 @@
 	var/obj/item/weapon/disk/data/diskette = null //Mostly so the geneticist can steal everything.
 	var/loading = 0 // Nice loading text
 	var/autoprocess = 0
-	var/data[0]
+	var/obj/machinery/clonepod/selected_pod
 
 	light_color = LIGHT_COLOR_DARKBLUE
 
-/obj/machinery/computer/cloning/New()
+/obj/machinery/computer/cloning/initialize()
+	..()
+	updatemodules()
+	
+/obj/machinery/computer/cloning/Destroy()
+	releasecloner()
 	..()
-	spawn(5)
-		updatemodules()
-		return
-	return
 
 /obj/machinery/computer/cloning/process()
-	if(!(scanner && pod1 && autoprocess))
+	if(!scanner || !pods.len || !autoprocess)
 		return
 
 	if(scanner.occupant && (scanner.scan_level > 2))
 		scan_mob(scanner.occupant)
 
-	if(!(pod1.occupant || pod1.mess) && (pod1.efficiency > 5))
-		for(var/datum/dna2/record/R in src.records)
-			if(!(pod1.occupant || pod1.mess))
-				if(pod1.growclone(R))
-					records.Remove(R)
-
+	for (var/obj/machinery/clonepod/pod in pods)
+		if(!(pod.occupant || pod.mess) && (pod.efficiency > 5))
+			for(var/datum/dna2/record/R in src.records)
+				if(!(pod.occupant || pod.mess))
+					if(pod.growclone(R))
+						records.Remove(R)		
+						
 /obj/machinery/computer/cloning/proc/updatemodules()
 	src.scanner = findscanner()
-	src.pod1 = findcloner()
-
-	if (!isnull(src.pod1))
-		src.pod1.connected = src // Some variable the pod needs
+	releasecloner()
+	findcloner()
+	if(!selected_pod)
+		selected_pod = pods[1]
 
 /obj/machinery/computer/cloning/proc/findscanner()
 	var/obj/machinery/dna_scannernew/scannerf = null
 
-	// Loop through every direction
+	//Try to find scanner on adjacent tiles first
 	for(dir in list(NORTH,EAST,SOUTH,WEST))
-
-		// Try to find a scanner in that direction
 		scannerf = locate(/obj/machinery/dna_scannernew, get_step(src, dir))
+		if (scannerf)
+			return scannerf
 
-		// If found, then we break, and return the scanner
-		if (!isnull(scannerf))
-			break
-
-	// If no scanner was found, it will return null
-	return scannerf
-
-/obj/machinery/computer/cloning/proc/findcloner()
-	var/obj/machinery/clonepod/podf = null
+	//Then look for a free one in the area
+	if(!scannerf)
+		for(var/obj/machinery/dna_scannernew/S in get_area(src))
```

Comment:
I don't want to say that being able to automatically connect to any scanner in the area is a __bad__ thing, but, it's a bit odd. Might cause unintended issues with a console not being fully constructed.